bin/improver-nbhood-iterate-with-mask

```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# (C) British Crown Copyright 2017 Met Office.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Script to run neighbourhooding processing when iterating over a coordinate
+defining a series of masks."""
+
+
+import iris
+
+from improver.argparser import ArgParser
+from improver.nbhood.use_nbhood import ApplyNeighbourhoodProcessingWithAMask
+
+
+def main():
+    """Load in arguments for applying neighbourhood processing when using a
+    mask."""
+    parser = ArgParser(
+        description='Apply the requested neighbourhood method via '
+                    'the ApplyNeighbourhoodProcessingWithAMask plugin '
+                    'to a file with one cube using a mask. Within the mask '
```

Comment:
Change this to something like:
"'Apply the requested neighbourhood method via the ApplyNeighbourhoodProcessingWithAMask plugin to a file with one cube using a mask. The cube containing the mask may have an extra dimension compared to the input cube. In this case the extra coordinate is iterated over and each mask is applied to the input cube separately. These masked cubes are concatenated together in the output cube, resulting in a cube that has been processed using multiple masks, with the extra dimension from the mask cube. There is also an option to re-mask...."